### ISSUE TITLE
Startup crash from having the mods folder as a symlink destination

### ISSUE THREAD

**Issue by mosharky** [Sinytra/ConnectorExtras#12]
### Describe the bug

A startup crash occurs due to "Missing or unsupported mandatory dependencies", where Connector Extras isn't detecting Connector.

### Steps to reproduce

1. Create a mods folder as a symbolic link, such that your launchers instance has the destination of the link
2. Add Sinytra Connector, Connector Extras, and Forgified Fabric API
3. Launch the instance
4. Startup crash will occur with error: _"Missing or unsupported mandatory dependencies: Mod ID: 'connectormod', Requested by: 'connectorextras_architectury_bridge', Expected range: '[1.0.0-beta.18,)', Actual version: '[MISSING]'"_

### Logs

[latest.log](https://mclo.gs/7sBN6F9)

### Additional context

- Seems like Connector Extras isn't detecting Connector properly
- This was tested on Prism Launcher and CurseForge
- Only Connector Extras and its dependencies were used in the test
- The crash doesn't occur if my instance's mods folder is the source of a symlink

If you're wondering why I'd want to have a symlink, its because Prism can't export in the curse modpack format, and that adding/updating mods is a hassle with Prism, so I'm using CurseForge to manage the mods.

**Comment by Su5eD** [comment]
Does this work with mods other than Connector? The error comes from FML'd dependency resolver, and Connector Extras doesn't come with any custom logic for locating dependencies.

**Comment by mosharky** [comment]
Yup, I tested it with Moonlight Lib + Supplementaries. No crash.

**Comment by 8MiYile** [comment]
I'm having one of those problems.
